src/components/Calculator.js

```diff
@@ -1,59 +1,50 @@
-import React from 'react';
+import React, { useState } from 'react';
 import calculate from '../logic/calculate';
 
-class Calculator extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      total: '0',
-      next: '',
-      operation: '',
-    };
-    this.eventClicked = this.eventClicked.bind(this);
-  }
+function Calculator() {
+  const [input, setInput] = useState({ total: 0, next: null, operation: null });
 
-  eventClicked(e) {
-    const { total, next, operation } = this.state;
-    this.setState(calculate({ total, next, operation }, e.target.textContent));
-  }
+  const eventClicked = (e) => {
+    const test = { ...input, ...calculate(input, e.target.textContent) };
+    setInput(() => ({ ...test }));
+    // const { total, next, operation } = this.state;
+    // this.setState(calculate({ total, next, operation }, e.target.textContent));
```

Comment:
**Changes Required**
- [ ] Commented out code makes our code base look bad. We can use `git history` if we need to check the changelog. Let us try and tidy up the codebase by removing these comments.